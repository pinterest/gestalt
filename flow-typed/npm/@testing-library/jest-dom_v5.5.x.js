// flow-typed signature: 5291c1af24f46ab1b593d2626cfbb310
// flow-typed version: 7ad57fc095/@testing-library/jest-dom_v5.5.x/flow_>=v0.104.x

declare module '@testing-library/jest-dom' {
  declare type JestMatcherResult = {
    message?: string | (() => string),
    pass: boolean,
    ...
  };

  declare type Result = JestMatcherResult | Promise<JestMatcherResult>;

  declare module.exports: {|
    /**
     * @deprecated
     */
    toBeInTheDOM(container?: HTMLElement): Result,

    toBeInTheDocument(): Result,
    toBeVisible(): Result,
    toBeEmpty(): Result,
    toBeDisabled(): Result,
    toBeEnabled(): Result,
    toBeInvalid(): Result,
    toBeRequired(): Result,
    toBeValid(): Result,
    toContainElement(element: HTMLElement | null): Result,
    toContainHTML(htmlText: string): Result,
    toHaveAttribute(attr: string, value?: any): Result,
    toHaveClass(...classNames: string[]): Result,
    toHaveFocus(): Result,
    toHaveFormValues(expectedValues: { [name: string]: any, ... }): Result,
    toHaveStyle(css: string | { [name: string]: any, ... }): Result,
    toHaveTextContent(
      text: string | RegExp,
      options?: {| normalizeWhitespace: boolean |}
    ): Result,
    toHaveValue(value?: string | string[] | number): Result,
    toHaveDisplayValue(value: string | string[]): Result,
    toBeChecked(): Result,
  |};
}

declare module '@testing-library/jest-dom/extend-expect' {
  declare module.exports: any;
}
