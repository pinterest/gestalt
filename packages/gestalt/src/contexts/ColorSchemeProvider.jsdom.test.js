// @flow strict
import { act, render, screen } from '@testing-library/react';
import ColorSchemeProvider, { useColorScheme } from './ColorSchemeProvider.js';

function ThemeAwareComponent() {
  const theme = useColorScheme();
  return <div>{theme.name}</div>;
}

describe('ColorSchemeProvider', () => {
  it('renders child content in a div', () => {
    const { container } = render(<ColorSchemeProvider>Child 1</ColorSchemeProvider>);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('div')).toMatchInlineSnapshot(`
        <div
          class=""
        >
          Child 1
        </div>
      `);
  });

  it('renders styling for light mode when no color scheme specified', () => {
    const { container } = render(<ColorSchemeProvider>Content</ColorSchemeProvider>);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('style')).toMatchSnapshot();
  });

  it('renders styling for light mode when specified', () => {
    const { container } = render(
      <ColorSchemeProvider colorScheme="light">Content</ColorSchemeProvider>,
    );
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('style')).toMatchSnapshot();
  });

  it('renders styling for dark mode when specified', () => {
    const { container } = render(
      <ColorSchemeProvider colorScheme="dark">Content</ColorSchemeProvider>,
    );
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('style')).toMatchSnapshot();
  });

  it('renders styling with media query when userPreference', () => {
    const { container } = render(
      <ColorSchemeProvider colorScheme="userPreference">Content</ColorSchemeProvider>,
    );
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('style')).toMatchSnapshot();
  });

  it('renders styling with a custom class if has an id', () => {
    const { container } = render(<ColorSchemeProvider id="testId">Content</ColorSchemeProvider>);
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('.__gestaltThemetestId')).toBeTruthy();
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    expect(container.querySelector('style')).toMatchSnapshot();
  });
});

describe('useColorScheme', () => {
  it('uses light mode theme when not in theme provider', () => {
    render(<ThemeAwareComponent />);
    expect(screen.getByText('lightMode')).toBeTruthy();
  });

  it('uses light mode theme when specified', () => {
    render(
      <ColorSchemeProvider colorScheme="light">
        <ThemeAwareComponent />
      </ColorSchemeProvider>,
    );
    expect(screen.getByText('lightMode')).toBeTruthy();
  });

  it('uses dark mode theme when specified', () => {
    render(
      <ColorSchemeProvider colorScheme="dark">
        <ThemeAwareComponent />
      </ColorSchemeProvider>,
    );
    expect(screen.getByText('darkMode')).toBeTruthy();
  });

  it('uses theme based on matchMedia when userPreference', () => {
    // $FlowFixMe[unclear-type]
    let listener = jest.fn<[{ matches: boolean }], any>();
    window.matchMedia = () => ({
      addListener: (cb) => {
        listener = cb;
      },
      removeListener: jest.fn(),
    });
    render(
      <ColorSchemeProvider colorScheme="userPreference">
        <ThemeAwareComponent />
      </ColorSchemeProvider>,
    );
    expect(screen.getByText('lightMode')).toBeTruthy();
    act(() => listener({ matches: true }));
    expect(screen.getByText('darkMode')).toBeTruthy();
  });
});
