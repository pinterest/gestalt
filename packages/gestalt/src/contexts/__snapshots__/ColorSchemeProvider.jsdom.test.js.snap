// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSchemeProvider renders styling for dark mode when specified 1`] = `
<style>
  :root {
  --g-colorRed0: #e60023;
  --g-colorRed100: #ff5247;
  --g-colorRed100Active: #b8001b;
  --g-colorRed100Hovered: #cf001f;
  --g-colorGray0: #030303;
  --g-colorGray0Active: #1f1f1f;
  --g-colorGray0Hovered: #121212;
  --g-colorGray50: #212121;
  --g-colorGray100: #404040;
  --g-colorGray100Active: #666;
  --g-colorGray100Hovered: #535353;
  --g-colorGray150: #585858;
  --g-colorGray150Hovered: #535353;
  --g-colorGray200: #ababab;
  --g-colorGray200Active: #9b9b9b;
  --g-colorGray200Hovered: #919191;
  --g-colorGray300: #efefef;
  --g-colorGray400: #fff;
  --g-colorTransparentDarkGray: rgba(255, 255, 255, 0.8);
  --g-colorTransparentGray60: rgba(250, 250, 250, 0.5);
  --g-colorTransparentGray100: rgba(250, 250, 250, 0.6);
  --g-colorTransparentGray500: rgba(0, 0, 0, 0.5);
  --g-colorTransparentWhite: rgba(51, 51, 51, 0.8);
  --g-colorDataVisualization10: #007A72;
  --g-colorDataVisualization11: #f76593;
  --g-colorDataVisualization12: #ffc58f;
  --g-colorDataVisualizationPrimary: #75bfff;
  --g-colorDataVisualizationSuccessGraph: #6bec8c;
  --g-colorDataVisualizationSuccessText: #39d377;
  --g-colorDataVisualizationErrorGraph: #f47171;
  --g-colorDataVisualizationErrorText: #eb4242;
  --g-colorDataVisualization01: #005fcb;
  --g-colorDataVisualization02: #75e4d5;
  --g-colorDataVisualization03: #B190FF;
  --g-colorDataVisualization04: #FDA600;
  --g-colorDataVisualization05: #75bfff;
  --g-colorDataVisualization06: #de2c62;
  --g-colorDataVisualization07: #a4f9ac;
  --g-colorDataVisualization08: #812ae7;
  --g-colorDataVisualization09: #ff5b45;
  --color-text-default: #ffffff;
  --color-text-subtle: #a5a5a5;
  --color-text-success: #39d377;
  --color-text-disabled: #4a4a4a;
  --color-text-error: #f47171;
  --color-text-warning: #e18d00;
  --color-text-inverse: #111111;
  --color-text-shopping: #75bfff;
  --color-text-link: #45a3fe;
  --color-text-icon-default: #ffffff;
  --color-text-icon-subtle: #a5a5a5;
  --color-text-icon-success: #39d377;
  --color-text-icon-disabled: #4a4a4a;
  --color-text-icon-error: #f47171;
  --color-text-icon-warning: #e18d00;
  --color-text-icon-info: #75bfff;
  --color-text-icon-recommendation: #b190ff;
  --color-text-icon-inverse: #111111;
  --color-text-icon-shopping: #75bfff;
  --color-background-default: #111111;
  --color-background-info-base: #75bfff;
  --color-background-info-weak: #003c96;
  --color-background-info-strong: #abdbff;
  --color-background-error-base: #f47171;
  --color-background-error-weak: #660000;
  --color-background-warning-base: #fdc900;
  --color-background-warning-weak: #7c2d00;
  --color-background-success-base: #6bec8c;
  --color-background-success-weak: #00422c;
  --color-background-recommendation-base: #b190ff;
  --color-background-recommendation-weak: #550aa9;
  --color-background-shopping: #75bfff;
  --color-background-secondary-base: #767676;
  --color-background-tertiary-base: #cdcdcd;
  --color-background-selected-base: #e9e9e9;
  --color-background-inverse-base: #f9f9f9;
  --color-background-education: #75bfff;
  --color-background-elevation-accent: #191919;
  --color-background-elevation-floating: #2b2b2b;
  --color-background-elevation-raised: #4a4a4a;
  --color-border-container: #767676;
  --color-border-default: #cdcdcd;
  --color-border-error: #f47171;
  --color-data-visualization-10: #007A72;
  --color-data-visualization-11: #f76593;
  --color-data-visualization-12: #ffc58f;
  --color-data-visualization-primary: #75bfff;
  --color-data-visualization-success-graph: #6bec8c;
  --color-data-visualization-success-text: #39d377;
  --color-data-visualization-error-graph: #f47171;
  --color-data-visualization-error-text: #eb4242;
  --color-data-visualization-01: #005fcb;
  --color-data-visualization-02: #75e4d5;
  --color-data-visualization-03: #B190FF;
  --color-data-visualization-04: #FDA600;
  --color-data-visualization-05: #75bfff;
  --color-data-visualization-06: #de2c62;
  --color-data-visualization-07: #a4f9ac;
  --color-data-visualization-08: #812ae7;
  --color-data-visualization-09: #ff5b45;
  --elevation-floating: none;
  --elevation-raised-top: 0 0.5px 0 #f9f9f9;
  --elevation-raised-bottom: 0 -0.5px 0 #f9f9f9;
 }
</style>
`;

exports[`ColorSchemeProvider renders styling for light mode when no color scheme specified 1`] = `
<style>
  :root {
  --g-colorRed0: #ff5247;
  --g-colorRed100: #e60023;
  --g-colorRed100Active: #a3081a;
  --g-colorRed100Hovered: #ad081b;
  --g-colorGray0: #fff;
  --g-colorGray0Active: #e0e0e0;
  --g-colorGray0Hovered: #f0f0f0;
  --g-colorGray50: #fff;
  --g-colorGray100: #efefef;
  --g-colorGray100Active: #dadada;
  --g-colorGray100Hovered: #e2e2e2;
  --g-colorGray150: #ddd;
  --g-colorGray150Hovered: #d0d0d0;
  --g-colorGray200: #767676;
  --g-colorGray200Active: #828282;
  --g-colorGray200Hovered: #878787;
  --g-colorGray300: #111;
  --g-colorGray400: #000;
  --g-colorTransparentDarkGray: rgba(51, 51, 51, 0.8);
  --g-colorTransparentGray60: rgba(0, 0, 0, 0.06);
  --g-colorTransparentGray100: rgba(0, 0, 0, 0.1);
  --g-colorTransparentGray500: rgba(0, 0, 0, 0.1);
  --g-colorTransparentWhite: rgba(255, 255, 255, 0.8);
  --g-colorDataVisualization10: #005062;
  --g-colorDataVisualization11: #de2c62;
  --g-colorDataVisualization12: #660e00;
  --g-colorDataVisualizationPrimary: #0081fe;
  --g-colorDataVisualizationSuccessGraph: #008753;
  --g-colorDataVisualizationSuccessText: #005f3e;
  --g-colorDataVisualizationErrorGraph: #ed0000;
  --g-colorDataVisualizationErrorText: #cc0000;
  --g-colorDataVisualization01: #003c96;
  --g-colorDataVisualization02: #11a69c;
  --g-colorDataVisualization03: #924af7;
  --g-colorDataVisualization04: #d17711;
  --g-colorDataVisualization05: #0081fe;
  --g-colorDataVisualization06: #ff5383;
  --g-colorDataVisualization07: #00ab55;
  --g-colorDataVisualization08: #400387;
  --g-colorDataVisualization09: #f2681f;
  --color-text-default: #111111;
  --color-text-subtle: #5f5f5f;
  --color-text-success: #005f3e;
  --color-text-disabled: #a5a5a5;
  --color-text-error: #cc0000;
  --color-text-warning: #bd5b00;
  --color-text-inverse: #ffffff;
  --color-text-shopping: #0074e8;
  --color-text-link: #004ba9;
  --color-text-icon-default: #111111;
  --color-text-icon-subtle: #5f5f5f;
  --color-text-icon-success: #005f3e;
  --color-text-icon-disabled: #a5a5a5;
  --color-text-icon-error: #cc0000;
  --color-text-icon-warning: #bd5b00;
  --color-text-icon-info: #0074e8;
  --color-text-icon-recommendation: #812ae7;
  --color-text-icon-inverse: #ffffff;
  --color-text-icon-shopping: #0074e8;
  --color-background-default: #ffffff;
  --color-background-info-base: #0074e8;
  --color-background-info-weak: #d7edff;
  --color-background-info-strong: #005fcb;
  --color-background-error-base: #cc0000;
  --color-background-error-weak: #ffe0e0;
  --color-background-warning-base: #bd5b00;
  --color-background-warning-weak: #ffe4c1;
  --color-background-success-base: #008753;
  --color-background-success-weak: #c3f9c2;
  --color-background-recommendation-base: #812ae7;
  --color-background-recommendation-weak: #e9e4ff;
  --color-background-shopping: #0074e8;
  --color-background-secondary-base: #e9e9e9;
  --color-background-tertiary-base: #767676;
  --color-background-selected-base: #111111;
  --color-background-inverse-base: #111111;
  --color-background-education: #0074e8;
  --color-background-elevation-accent: #f1f1f1;
  --color-background-elevation-floating: rgba(0, 0, 0, 0);
  --color-background-elevation-raised: rgba(0, 0, 0, 0);
  --color-border-container: #cdcdcd;
  --color-border-default: #767676;
  --color-border-error: #cc0000;
  --color-data-visualization-10: #005062;
  --color-data-visualization-11: #de2c62;
  --color-data-visualization-12: #660e00;
  --color-data-visualization-primary: #0081fe;
  --color-data-visualization-success-graph: #008753;
  --color-data-visualization-success-text: #005f3e;
  --color-data-visualization-error-graph: #ed0000;
  --color-data-visualization-error-text: #cc0000;
  --color-data-visualization-01: #003c96;
  --color-data-visualization-02: #11a69c;
  --color-data-visualization-03: #924af7;
  --color-data-visualization-04: #d17711;
  --color-data-visualization-05: #0081fe;
  --color-data-visualization-06: #ff5383;
  --color-data-visualization-07: #00ab55;
  --color-data-visualization-08: #400387;
  --color-data-visualization-09: #f2681f;
  --elevation-floating: 0 0 8px rgba(0, 0, 0, 0.10);
  --elevation-raised-top: 0 2px 8px rgba(0, 0, 0, 0.12);
  --elevation-raised-bottom: 0 -2px 8px rgba(0, 0, 0, 0.12);
 }
</style>
`;

exports[`ColorSchemeProvider renders styling for light mode when specified 1`] = `
<style>
  :root {
  --g-colorRed0: #ff5247;
  --g-colorRed100: #e60023;
  --g-colorRed100Active: #a3081a;
  --g-colorRed100Hovered: #ad081b;
  --g-colorGray0: #fff;
  --g-colorGray0Active: #e0e0e0;
  --g-colorGray0Hovered: #f0f0f0;
  --g-colorGray50: #fff;
  --g-colorGray100: #efefef;
  --g-colorGray100Active: #dadada;
  --g-colorGray100Hovered: #e2e2e2;
  --g-colorGray150: #ddd;
  --g-colorGray150Hovered: #d0d0d0;
  --g-colorGray200: #767676;
  --g-colorGray200Active: #828282;
  --g-colorGray200Hovered: #878787;
  --g-colorGray300: #111;
  --g-colorGray400: #000;
  --g-colorTransparentDarkGray: rgba(51, 51, 51, 0.8);
  --g-colorTransparentGray60: rgba(0, 0, 0, 0.06);
  --g-colorTransparentGray100: rgba(0, 0, 0, 0.1);
  --g-colorTransparentGray500: rgba(0, 0, 0, 0.1);
  --g-colorTransparentWhite: rgba(255, 255, 255, 0.8);
  --g-colorDataVisualization10: #005062;
  --g-colorDataVisualization11: #de2c62;
  --g-colorDataVisualization12: #660e00;
  --g-colorDataVisualizationPrimary: #0081fe;
  --g-colorDataVisualizationSuccessGraph: #008753;
  --g-colorDataVisualizationSuccessText: #005f3e;
  --g-colorDataVisualizationErrorGraph: #ed0000;
  --g-colorDataVisualizationErrorText: #cc0000;
  --g-colorDataVisualization01: #003c96;
  --g-colorDataVisualization02: #11a69c;
  --g-colorDataVisualization03: #924af7;
  --g-colorDataVisualization04: #d17711;
  --g-colorDataVisualization05: #0081fe;
  --g-colorDataVisualization06: #ff5383;
  --g-colorDataVisualization07: #00ab55;
  --g-colorDataVisualization08: #400387;
  --g-colorDataVisualization09: #f2681f;
  --color-text-default: #111111;
  --color-text-subtle: #5f5f5f;
  --color-text-success: #005f3e;
  --color-text-disabled: #a5a5a5;
  --color-text-error: #cc0000;
  --color-text-warning: #bd5b00;
  --color-text-inverse: #ffffff;
  --color-text-shopping: #0074e8;
  --color-text-link: #004ba9;
  --color-text-icon-default: #111111;
  --color-text-icon-subtle: #5f5f5f;
  --color-text-icon-success: #005f3e;
  --color-text-icon-disabled: #a5a5a5;
  --color-text-icon-error: #cc0000;
  --color-text-icon-warning: #bd5b00;
  --color-text-icon-info: #0074e8;
  --color-text-icon-recommendation: #812ae7;
  --color-text-icon-inverse: #ffffff;
  --color-text-icon-shopping: #0074e8;
  --color-background-default: #ffffff;
  --color-background-info-base: #0074e8;
  --color-background-info-weak: #d7edff;
  --color-background-info-strong: #005fcb;
  --color-background-error-base: #cc0000;
  --color-background-error-weak: #ffe0e0;
  --color-background-warning-base: #bd5b00;
  --color-background-warning-weak: #ffe4c1;
  --color-background-success-base: #008753;
  --color-background-success-weak: #c3f9c2;
  --color-background-recommendation-base: #812ae7;
  --color-background-recommendation-weak: #e9e4ff;
  --color-background-shopping: #0074e8;
  --color-background-secondary-base: #e9e9e9;
  --color-background-tertiary-base: #767676;
  --color-background-selected-base: #111111;
  --color-background-inverse-base: #111111;
  --color-background-education: #0074e8;
  --color-background-elevation-accent: #f1f1f1;
  --color-background-elevation-floating: rgba(0, 0, 0, 0);
  --color-background-elevation-raised: rgba(0, 0, 0, 0);
  --color-border-container: #cdcdcd;
  --color-border-default: #767676;
  --color-border-error: #cc0000;
  --color-data-visualization-10: #005062;
  --color-data-visualization-11: #de2c62;
  --color-data-visualization-12: #660e00;
  --color-data-visualization-primary: #0081fe;
  --color-data-visualization-success-graph: #008753;
  --color-data-visualization-success-text: #005f3e;
  --color-data-visualization-error-graph: #ed0000;
  --color-data-visualization-error-text: #cc0000;
  --color-data-visualization-01: #003c96;
  --color-data-visualization-02: #11a69c;
  --color-data-visualization-03: #924af7;
  --color-data-visualization-04: #d17711;
  --color-data-visualization-05: #0081fe;
  --color-data-visualization-06: #ff5383;
  --color-data-visualization-07: #00ab55;
  --color-data-visualization-08: #400387;
  --color-data-visualization-09: #f2681f;
  --elevation-floating: 0 0 8px rgba(0, 0, 0, 0.10);
  --elevation-raised-top: 0 2px 8px rgba(0, 0, 0, 0.12);
  --elevation-raised-bottom: 0 -2px 8px rgba(0, 0, 0, 0.12);
 }
</style>
`;

exports[`ColorSchemeProvider renders styling with a custom class if has an id 1`] = `
<style>
  .__gestaltThemetestId {
  --g-colorRed0: #ff5247;
  --g-colorRed100: #e60023;
  --g-colorRed100Active: #a3081a;
  --g-colorRed100Hovered: #ad081b;
  --g-colorGray0: #fff;
  --g-colorGray0Active: #e0e0e0;
  --g-colorGray0Hovered: #f0f0f0;
  --g-colorGray50: #fff;
  --g-colorGray100: #efefef;
  --g-colorGray100Active: #dadada;
  --g-colorGray100Hovered: #e2e2e2;
  --g-colorGray150: #ddd;
  --g-colorGray150Hovered: #d0d0d0;
  --g-colorGray200: #767676;
  --g-colorGray200Active: #828282;
  --g-colorGray200Hovered: #878787;
  --g-colorGray300: #111;
  --g-colorGray400: #000;
  --g-colorTransparentDarkGray: rgba(51, 51, 51, 0.8);
  --g-colorTransparentGray60: rgba(0, 0, 0, 0.06);
  --g-colorTransparentGray100: rgba(0, 0, 0, 0.1);
  --g-colorTransparentGray500: rgba(0, 0, 0, 0.1);
  --g-colorTransparentWhite: rgba(255, 255, 255, 0.8);
  --g-colorDataVisualization10: #005062;
  --g-colorDataVisualization11: #de2c62;
  --g-colorDataVisualization12: #660e00;
  --g-colorDataVisualizationPrimary: #0081fe;
  --g-colorDataVisualizationSuccessGraph: #008753;
  --g-colorDataVisualizationSuccessText: #005f3e;
  --g-colorDataVisualizationErrorGraph: #ed0000;
  --g-colorDataVisualizationErrorText: #cc0000;
  --g-colorDataVisualization01: #003c96;
  --g-colorDataVisualization02: #11a69c;
  --g-colorDataVisualization03: #924af7;
  --g-colorDataVisualization04: #d17711;
  --g-colorDataVisualization05: #0081fe;
  --g-colorDataVisualization06: #ff5383;
  --g-colorDataVisualization07: #00ab55;
  --g-colorDataVisualization08: #400387;
  --g-colorDataVisualization09: #f2681f;
  --color-text-default: #111111;
  --color-text-subtle: #5f5f5f;
  --color-text-success: #005f3e;
  --color-text-disabled: #a5a5a5;
  --color-text-error: #cc0000;
  --color-text-warning: #bd5b00;
  --color-text-inverse: #ffffff;
  --color-text-shopping: #0074e8;
  --color-text-link: #004ba9;
  --color-text-icon-default: #111111;
  --color-text-icon-subtle: #5f5f5f;
  --color-text-icon-success: #005f3e;
  --color-text-icon-disabled: #a5a5a5;
  --color-text-icon-error: #cc0000;
  --color-text-icon-warning: #bd5b00;
  --color-text-icon-info: #0074e8;
  --color-text-icon-recommendation: #812ae7;
  --color-text-icon-inverse: #ffffff;
  --color-text-icon-shopping: #0074e8;
  --color-background-default: #ffffff;
  --color-background-info-base: #0074e8;
  --color-background-info-weak: #d7edff;
  --color-background-info-strong: #005fcb;
  --color-background-error-base: #cc0000;
  --color-background-error-weak: #ffe0e0;
  --color-background-warning-base: #bd5b00;
  --color-background-warning-weak: #ffe4c1;
  --color-background-success-base: #008753;
  --color-background-success-weak: #c3f9c2;
  --color-background-recommendation-base: #812ae7;
  --color-background-recommendation-weak: #e9e4ff;
  --color-background-shopping: #0074e8;
  --color-background-secondary-base: #e9e9e9;
  --color-background-tertiary-base: #767676;
  --color-background-selected-base: #111111;
  --color-background-inverse-base: #111111;
  --color-background-education: #0074e8;
  --color-background-elevation-accent: #f1f1f1;
  --color-background-elevation-floating: rgba(0, 0, 0, 0);
  --color-background-elevation-raised: rgba(0, 0, 0, 0);
  --color-border-container: #cdcdcd;
  --color-border-default: #767676;
  --color-border-error: #cc0000;
  --color-data-visualization-10: #005062;
  --color-data-visualization-11: #de2c62;
  --color-data-visualization-12: #660e00;
  --color-data-visualization-primary: #0081fe;
  --color-data-visualization-success-graph: #008753;
  --color-data-visualization-success-text: #005f3e;
  --color-data-visualization-error-graph: #ed0000;
  --color-data-visualization-error-text: #cc0000;
  --color-data-visualization-01: #003c96;
  --color-data-visualization-02: #11a69c;
  --color-data-visualization-03: #924af7;
  --color-data-visualization-04: #d17711;
  --color-data-visualization-05: #0081fe;
  --color-data-visualization-06: #ff5383;
  --color-data-visualization-07: #00ab55;
  --color-data-visualization-08: #400387;
  --color-data-visualization-09: #f2681f;
  --elevation-floating: 0 0 8px rgba(0, 0, 0, 0.10);
  --elevation-raised-top: 0 2px 8px rgba(0, 0, 0, 0.12);
  --elevation-raised-bottom: 0 -2px 8px rgba(0, 0, 0, 0.12);
 }
</style>
`;

exports[`ColorSchemeProvider renders styling with media query when userPreference 1`] = `
<style>
  @media(prefers-color-scheme: dark) {
  :root {
  --g-colorRed0: #e60023;
  --g-colorRed100: #ff5247;
  --g-colorRed100Active: #b8001b;
  --g-colorRed100Hovered: #cf001f;
  --g-colorGray0: #030303;
  --g-colorGray0Active: #1f1f1f;
  --g-colorGray0Hovered: #121212;
  --g-colorGray50: #212121;
  --g-colorGray100: #404040;
  --g-colorGray100Active: #666;
  --g-colorGray100Hovered: #535353;
  --g-colorGray150: #585858;
  --g-colorGray150Hovered: #535353;
  --g-colorGray200: #ababab;
  --g-colorGray200Active: #9b9b9b;
  --g-colorGray200Hovered: #919191;
  --g-colorGray300: #efefef;
  --g-colorGray400: #fff;
  --g-colorTransparentDarkGray: rgba(255, 255, 255, 0.8);
  --g-colorTransparentGray60: rgba(250, 250, 250, 0.5);
  --g-colorTransparentGray100: rgba(250, 250, 250, 0.6);
  --g-colorTransparentGray500: rgba(0, 0, 0, 0.5);
  --g-colorTransparentWhite: rgba(51, 51, 51, 0.8);
  --g-colorDataVisualization10: #007A72;
  --g-colorDataVisualization11: #f76593;
  --g-colorDataVisualization12: #ffc58f;
  --g-colorDataVisualizationPrimary: #75bfff;
  --g-colorDataVisualizationSuccessGraph: #6bec8c;
  --g-colorDataVisualizationSuccessText: #39d377;
  --g-colorDataVisualizationErrorGraph: #f47171;
  --g-colorDataVisualizationErrorText: #eb4242;
  --g-colorDataVisualization01: #005fcb;
  --g-colorDataVisualization02: #75e4d5;
  --g-colorDataVisualization03: #B190FF;
  --g-colorDataVisualization04: #FDA600;
  --g-colorDataVisualization05: #75bfff;
  --g-colorDataVisualization06: #de2c62;
  --g-colorDataVisualization07: #a4f9ac;
  --g-colorDataVisualization08: #812ae7;
  --g-colorDataVisualization09: #ff5b45;
  --color-text-default: #ffffff;
  --color-text-subtle: #a5a5a5;
  --color-text-success: #39d377;
  --color-text-disabled: #4a4a4a;
  --color-text-error: #f47171;
  --color-text-warning: #e18d00;
  --color-text-inverse: #111111;
  --color-text-shopping: #75bfff;
  --color-text-link: #45a3fe;
  --color-text-icon-default: #ffffff;
  --color-text-icon-subtle: #a5a5a5;
  --color-text-icon-success: #39d377;
  --color-text-icon-disabled: #4a4a4a;
  --color-text-icon-error: #f47171;
  --color-text-icon-warning: #e18d00;
  --color-text-icon-info: #75bfff;
  --color-text-icon-recommendation: #b190ff;
  --color-text-icon-inverse: #111111;
  --color-text-icon-shopping: #75bfff;
  --color-background-default: #111111;
  --color-background-info-base: #75bfff;
  --color-background-info-weak: #003c96;
  --color-background-info-strong: #abdbff;
  --color-background-error-base: #f47171;
  --color-background-error-weak: #660000;
  --color-background-warning-base: #fdc900;
  --color-background-warning-weak: #7c2d00;
  --color-background-success-base: #6bec8c;
  --color-background-success-weak: #00422c;
  --color-background-recommendation-base: #b190ff;
  --color-background-recommendation-weak: #550aa9;
  --color-background-shopping: #75bfff;
  --color-background-secondary-base: #767676;
  --color-background-tertiary-base: #cdcdcd;
  --color-background-selected-base: #e9e9e9;
  --color-background-inverse-base: #f9f9f9;
  --color-background-education: #75bfff;
  --color-background-elevation-accent: #191919;
  --color-background-elevation-floating: #2b2b2b;
  --color-background-elevation-raised: #4a4a4a;
  --color-border-container: #767676;
  --color-border-default: #cdcdcd;
  --color-border-error: #f47171;
  --color-data-visualization-10: #007A72;
  --color-data-visualization-11: #f76593;
  --color-data-visualization-12: #ffc58f;
  --color-data-visualization-primary: #75bfff;
  --color-data-visualization-success-graph: #6bec8c;
  --color-data-visualization-success-text: #39d377;
  --color-data-visualization-error-graph: #f47171;
  --color-data-visualization-error-text: #eb4242;
  --color-data-visualization-01: #005fcb;
  --color-data-visualization-02: #75e4d5;
  --color-data-visualization-03: #B190FF;
  --color-data-visualization-04: #FDA600;
  --color-data-visualization-05: #75bfff;
  --color-data-visualization-06: #de2c62;
  --color-data-visualization-07: #a4f9ac;
  --color-data-visualization-08: #812ae7;
  --color-data-visualization-09: #ff5b45;
  --elevation-floating: none;
  --elevation-raised-top: 0 0.5px 0 #f9f9f9;
  --elevation-raised-bottom: 0 -0.5px 0 #f9f9f9;
 }
}
</style>
`;
